src/radical/saga/utils/pty_process.py

```diff
@@ -97,16 +98,22 @@ def __init__ (self, command, cfg=None, logger=None) :
         :param logger:  logger stream to send status messages to.
         """
 
-        if cfg:
-            self._cfg = cfg
-        else:
-            self._cfg = ru.Config('radical.saga', 'utils')['pty']
-
+        self.rlock  = mt.RLock()
         self.logger = logger
+
         if  not  self.logger : self.logger = ru.Logger('radical.saga.pty')
         self.logger.debug ("PTYProcess init %s" % self)
 
 
+        name = None
+        if isinstance(cfg, str):
+            name = cfg
+            cfg  = None
+
+        self.cfg = ru.Config('radical.saga.session', name=name, cfg=cfg)
+        self.cfg = self.cfg.pty
```

Comment:
I see this as a dependency loop. `name` is set as the cfg. The `cfg` is then setup as an `ru.Config`, which the is changed to a specific entry of the cfg dict. Why all these? Would it make more sense to have the name as a constructor parameter, as well as, the config? Name can be `'utils'` and cfg can be `None` by default.

What do you think?

Reply:
That would make perfect sense, and I pondered doing so, but decided to keep the API stable and overload the `cfg` parameter.  We do the same (for good or for worse) in other places in RP, where a `cfg` can be an explicit cnofig dictionary or `ru.Config` instance, or can be a string which points to a know config.  I agree that we may want to clean this up at some point, but I'm not sure if we should right now?

Reply:
I would like to merge this ASAP as it is now possibly blocking a user (Eugene). Could you open a ticket about the cleanup so that it stays on our radar/roadmap?

Reply:
I agree with @mturilli and open an issue for this.